repose-aggregator/core/repose-core-api/src/test/scala/org/openrepose/core/container/config/ContainerSchemaTest.scala

```diff
@@ -166,55 +212,102 @@ class ContainerSchemaTest extends FunSpec with Matchers {
+    it("should accept a config where the Via Configuration is defined at both the deployment and cluster config elements") {
       val config =
         """<repose-container xmlns='http://docs.openrepose.org/repose/container/v2.0'>
           |    <deployment-config>
           |        <deployment-directory>/var/repose</deployment-directory>
           |        <artifact-directory>/usr/share/repose/filters</artifact-directory>
+          |        <via-configuration content="test"/>
           |    </deployment-config>
           |
           |    <cluster-config cluster-id="foo">
-          |        <ssl-configuration>
-          |            <key-password>password</key-password>
-          |        </ssl-configuration>
+          |        <via-configuration content="override"/>
           |    </cluster-config>
           |</repose-container>""".stripMargin
-      val exception = intercept[SAXParseException] {
-        validator.validateConfigString(config)
-      }
-      exception.getLocalizedMessage should include("When defining a new keystore to use, the keystore and key passwords are required")
+      validator.validateConfigString(config)
     }
 
-    it("should successfully validate a config where ssl-configuration is partially patched and is in the base") {
+    it("should accept a config where the Via Configuration disables the via header") {
       val config =
         """<repose-container xmlns='http://docs.openrepose.org/repose/container/v2.0'>
           |    <deployment-config>
           |        <deployment-directory>/var/repose</deployment-directory>
           |        <artifact-directory>/usr/share/repose/filters</artifact-directory>
-          |        <ssl-configuration>
-          |            <keystore-filename>keystore.jks</keystore-filename>
-          |            <keystore-password>password</keystore-password>
-          |            <key-password>password</key-password>
-          |        </ssl-configuration>
           |    </deployment-config>
           |
           |    <cluster-config cluster-id="foo">
-          |        <ssl-configuration>
-          |            <key-password>other-password</key-password>
-          |        </ssl-configuration>
+          |        <via-configuration enabled="false"/>
           |    </cluster-config>
           |</repose-container>""".stripMargin
       validator.validateConfigString(config)
     }
   }
+
+  private val coreSpringProvider = CoreSpringProvider.getInstance()
+  coreSpringProvider.initializeCoreContext("/etc/repose", false)
+  private val reposeVersion = coreSpringProvider.getCoreContext.getEnvironment.getProperty(
+    ReposeSpringProperties.stripSpringValueStupidity(ReposeSpringProperties.CORE.REPOSE_VERSION))
+  describe("deprecated schema validation") {
+    if (reposeVersion.startsWith("8.")) {
+      it("should reject a config where cluster IDs are not unique across all patches") {
```

Comment:
Sure, but the thing being tested is tested elsewhere. Sure, the `via` attribute is used, but it's just sitting there as extraneous text.

Also, I'm not sure we need or even want to have unit tests for every XML node we define. Enforcing those constraints is on the validation engine. Otherwise we end up in the situation where we are testing every node that exists. Perhaps even every permutation of every node. Perhaps even nodes that are not supposed to exist. That is a lot of testing to double check the validation engine.

In my mind, what we absolutely ought to test is our assertions, since those are:
1. Potentially complex.
1. Enforcing arbitrary logic.

The rest just needs to be declared correctly.

Although, testing that we can validate a full example configuration is probably legit, just to prove that user configuration will look the way we expect it to. Also, testing the removal of things is legit, too, to prove that things that were are no longer.

That's how I see it, anyway.

/me Gets off high horse.